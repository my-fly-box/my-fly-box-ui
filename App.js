import React, { Component } from "react";
import { StyleSheet } from "react-native";
import { createMaterialBottomTabNavigator } from "@react-navigation/material-bottom-tabs";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import FlyBox from "./containers/FlyBox";
import AddFly from "./containers/AddFly";
import EditFly from "./containers/EditFly";
import FishCaught from "./containers/FishCaught";
import AddFish from "./containers/AddFish";
import EditFish from "./containers/EditFish";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./reducers";
import PhotoSelector from "./components/PhotoSelector";

const Tab = createMaterialBottomTabNavigator();
const FlyBoxStack = createStackNavigator();
const AddFlyStack = createStackNavigator();
const FishStack = createStackNavigator();
const store = createStore(rootReducer, composeWithDevTools());

function MyTabs() {
  return (
    <Tab.Navigator
      initialRouteName="MyFlyBox"
      activeColor={"#f7841f"}
      barStyle={{ backgroundColor: "#212326" }}
    >
      <Tab.Screen
        name="MyFlyBox"
        component={FlyBoxContainer}
        options={{
          tabBarLabel: "My Fly Box",
          tabBarIcon: ({ color }) => (
            <MaterialCommunityIcons name="hook" color={color} size={25} />
          ),
        }}
      />
      <Tab.Screen
        name="AddFly"
        component={AddFlyContainer}
        options={{
          tabBarLabel: "Add Fly",
          tabBarIcon: ({ color }) => (
            <MaterialCommunityIcons name="plus" color={color} size={25} />
          ),
        }}
      />
      <Tab.Screen
        name="FishCaught"
        component={FishContainer}
        options={{
          tabBarLabel: "Fish Caught",
          tabBarIcon: ({ color }) => (
            <MaterialCommunityIcons name="fish" color={color} size={25} />
          ),
        }}
      />
    </Tab.Navigator>
  );
}

function FlyBoxContainer() {
  return (
    <FlyBoxStack.Navigator>
      <FlyBoxStack.Screen
        name="MyFlyBox"
        component={FlyBox}
        options={{
          title: "My Fly Box",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
      <FlyBoxStack.Screen
        name="EditFly"
        component={EditFly}
        options={{
          title: "Edit Fly",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
    </FlyBoxStack.Navigator>
  );
}

function AddFlyContainer() {
  return (
    <AddFlyStack.Navigator>
      <AddFlyStack.Screen
        name="AddFly"
        component={AddFly}
        options={{
          title: "Add New Fly",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
    </AddFlyStack.Navigator>
  );
}

function FishContainer() {
  return (
    <FishStack.Navigator>
      <FishStack.Screen
        name="FishCaught"
        component={FishCaught}
        options={{
          title: "Fish Caught",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
      <FishStack.Screen
        name="EditFish"
        component={EditFish}
        options={{
          title: "Edit Fish",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
      <FishStack.Screen
        name="AddFish"
        component={AddFish}
        options={{
          title: "Add Fish",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
      <FishStack.Screen
        name="PhotoSelector"
        component={PhotoSelector}
        options={{
          title: "Upload or take a photo of your catch!",
          headerStyle: {
            backgroundColor: "#212326",
          },
          headerTintColor: "#ffffff",
          headerTitleStyle: {
            fontWeight: "bold",
          },
        }}
      />
    </FishStack.Navigator>
  );
}

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <NavigationContainer>
          <MyTabs />
        </NavigationContainer>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});

export default App;
