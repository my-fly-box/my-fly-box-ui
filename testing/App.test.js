import React from 'react';
import 'react-native'
import App from '../App'
import renderer from 'react-test-renderer'

describe('App', () => {
  it('renders the App', () => {
    const snap = renderer.create(
      <App />
    ).toJSON();
    expect(snap).toMatchSnapshot();
  })
})