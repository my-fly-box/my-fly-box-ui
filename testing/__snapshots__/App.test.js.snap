// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "rgb(242, 242, 242)",
        },
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      collapsable={false}
      importantForAccessibility="auto"
      pointerEvents="auto"
      removeClippedSubviews={false}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNCSafeAreaProvider
            onInsetsChange={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "zIndex": 1,
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityElementsHidden={false}
                importantForAccessibility="auto"
                onLayout={[Function]}
                pointerEvents="box-none"
                style={null}
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "zIndex": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#2A9D8F",
                        "borderBottomColor": "rgb(216, 216, 216)",
                        "flex": 1,
                        "shadowColor": "rgb(216, 216, 216)",
                        "shadowOffset": Object {
                          "height": 0.5,
                          "width": 0,
                        },
                        "shadowOpacity": 0.85,
                        "shadowRadius": 0,
                      }
                    }
                  />
                </View>
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "height": 64,
                      "maxHeight": undefined,
                      "minHeight": undefined,
                      "opacity": undefined,
                      "transform": undefined,
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "height": 20,
                      }
                    }
                  />
                  <View
                    pointerEvents="box-none"
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      pointerEvents="box-none"
                      style={
                        Object {
                          "marginHorizontal": 16,
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        accessibilityRole="header"
                        numberOfLines={1}
                        onLayout={[Function]}
                        style={
                          Object {
                            "color": "#fff",
                            "fontSize": 17,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        My Fly Box
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                />
                <View
                  accessibilityElementsHidden={false}
                  closing={false}
                  gestureVelocityImpact={0.3}
                  importantForAccessibility="auto"
                  onClose={[Function]}
                  onGestureBegin={[Function]}
                  onGestureCanceled={[Function]}
                  onOpen={[Function]}
                  onTransitionStart={[Function]}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  transitionSpec={
                    Object {
                      "close": Object {
                        "animation": "spring",
                        "config": Object {
                          "damping": 500,
                          "mass": 3,
                          "overshootClamping": true,
                          "restDisplacementThreshold": 10,
                          "restSpeedThreshold": 10,
                          "stiffness": 1000,
                        },
                      },
                      "open": Object {
                        "animation": "spring",
                        "config": Object {
                          "damping": 500,
                          "mass": 3,
                          "overshootClamping": true,
                          "restDisplacementThreshold": 10,
                          "restSpeedThreshold": 10,
                          "stiffness": 1000,
                        },
                      },
                    }
                  }
                >
                  <View
                    pointerEvents="box-none"
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      forwardedRef={[Function]}
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "transform": Array [
                            Object {
                              "translateX": 0,
                            },
                            Object {
                              "translateX": 0,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 1,
                              "width": -1,
                            },
                            "shadowOpacity": 0.3,
                            "shadowRadius": 5,
                            "top": 0,
                            "width": 3,
                          }
                        }
                      />
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "overflow": "hidden",
                            },
                            Array [
                              Object {
                                "backgroundColor": "rgb(242, 242, 242)",
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column-reverse",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "backgroundColor": "#E8E9F3",
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#BDC667",
                                    "borderBottomColor": "#2A9D8F",
                                    "borderBottomWidth": 2,
                                    "flexDirection": "row",
                                    "justifyContent": "space-around",
                                    "paddingVertical": 10,
                                    "width": "100%",
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#2A9D8F",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  Fly Name
                                </Text>
                                <Text
                                  style={
                                    Object {
                                      "color": "#2A9D8F",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  Fly Type
                                </Text>
                                <Text
                                  style={
                                    Object {
                                      "color": "#2A9D8F",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  Size
                                </Text>
                                <Text
                                  style={
                                    Object {
                                      "color": "#2A9D8F",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  Edit
                                </Text>
                                <Text
                                  style={
                                    Object {
                                      "color": "#2A9D8F",
                                      "fontSize": 18,
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  Delete
                                </Text>
                              </View>
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "justifyContent": "center",
                                    "paddingTop": 50,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontSize": 30,
                                      "fontWeight": "bold",
                                    }
                                  }
                                >
                                  Your fly box is empty.
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNCSafeAreaProvider>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "#2A9D8F",
        "bottom": 0,
        "elevation": 4,
        "left": 0,
        "position": null,
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2A9D8F",
          "overflow": "hidden",
        }
      }
    >
      <View
        forwardedRef={[Function]}
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flexDirection": "row",
            "maxWidth": 504,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "white",
                        },
                      ],
                    ]
                  }
                >
                  My Fly Box
                </Text>
              </View>
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "rgba(255, 255, 255, 0.5)",
                        },
                      ],
                    ]
                  }
                >
                  My Fly Box
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "white",
                        },
                      ],
                    ]
                  }
                >
                  Add Fly
                </Text>
              </View>
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "rgba(255, 255, 255, 0.5)",
                        },
                      ],
                    ]
                  }
                >
                  Add Fly
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontSize": 8,
                      "fontWeight": "400",
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "white",
                        },
                      ],
                    ]
                  }
                >
                  Fish Caught
                </Text>
              </View>
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontSize": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "rgba(255, 255, 255, 0.5)",
                        },
                      ],
                    ]
                  }
                >
                  Fish Caught
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
